app/components/For/index.js

```diff
@@ -7,23 +7,33 @@
 import React from 'react';
 import Proptypes from 'prop-types';
 import styled from 'styled-components';
+import If from '@components/If/';
 
 const FlexContainer = styled.div`
   display: flex;
   flex-direction: ${props => (props.isRow ? `row;` : `column;`)};
 `;
 
-export function For({ of, ParentComponent = props => <FlexContainer {...props} />, renderItem, noParent, ...props }) {
+export function For({
+  of = [],
+  ParentComponent = props => <FlexContainer {...props} />,
+  renderItem,
+  noParent,
+  ...props
+}) {
   const list = () => of.map((item, index) => ({ ...renderItem(item, index), key: index }));
   const children = () => (
     <ParentComponent {...props} data-testid="for">
       {list()}
     </ParentComponent>
   );
-  if (noParent) {
-    return (of || []).length ? list() : null;
-  }
-  return (of || []).length ? children() : null;
+  return (
+    <>
+      <If condition={noParent} otherwise={<>{children()}</>}>
+        <If condition={(of || []).length}>{list()}</If>
```

Comment:
Do we need `(of || [])` when we already have the default prop?

Reply:
@amitWaghamareWednesDay

Reply:
done